browser/playlist/test/playlist_download_request_manager_browsertests.cc

```diff
@@ -0,0 +1,218 @@
+            chrome_test_utils::GetProfile(this), component_manager_.get());
+  }
+
+  void TearDownOnMainThread() override {
+    request_manager_.reset();
+    component_manager_.reset();
+
+    ASSERT_TRUE(embedded_test_server()->ShutdownAndWaitUntilComplete());
+
+    PlatformBrowserTest::TearDownOnMainThread();
+  }
+
+ private:
+  std::unique_ptr<net::test_server::HttpResponse> Serve(
+      const std::string& html,
+      const net::test_server::HttpRequest& request) {
+    GURL absolute_url = embedded_test_server()->GetURL(request.relative_url);
+    if (absolute_url.path() != "/test")
+      return {};
+
+    auto response = std::make_unique<net::test_server::BasicHttpResponse>();
+    response->set_code(net::HTTP_OK);
+    response->set_content(html);
+    response->set_content_type("text/html; charset=utf-8");
+    return response;
+  }
+
+  void OnGetMedia(const char* test_name,
+                  const std::vector<playlist::PlaylistItemInfo>& expected_items,
+                  const std::vector<playlist::PlaylistItemInfo>& actual_items) {
+    VLOG(2) << test_name << ": " << __func__;
+
+    EXPECT_EQ(actual_items.size(), expected_items.size());
+
+    auto path_fixer = [this](auto& item) {
+      if (!item.media_file_path.empty()) {
+        item.media_file_path =
+            embedded_test_server()->GetURL(item.media_file_path).spec();
+      }
+
+      if (!item.thumbnail_path.empty()) {
+        item.thumbnail_path =
+            embedded_test_server()->GetURL(item.thumbnail_path).spec();
+      }
+    };
+    auto comparer = [](const auto& a, const auto& b) {
+      return a.media_file_path < b.media_file_path;
+    };
+    std::vector sorted_expected(expected_items.begin(), expected_items.end());
+    base::ranges::for_each(sorted_expected, path_fixer);
+    base::ranges::sort(sorted_expected, comparer);
+
+    std::vector sorted_actual(actual_items.begin(), actual_items.end());
+    base::ranges::sort(sorted_actual, comparer);
+
+    auto equal = [](const auto& a, const auto& b) {
+      // id is not compared because it is generated for |expected_items|.
+      return a.media_file_path == b.media_file_path && a.title == b.title &&
+             a.thumbnail_path == b.thumbnail_path;
+    };
+
+    if (!base::ranges::equal(sorted_expected, sorted_actual, equal)) {
+      std::string log("[");
+      log += test_name;
+      log += " Failed] Expected: [";
+      for (const auto& item : sorted_expected) {
+        log += "{ ";
+        log += item.media_file_path;
+        log += ", ";
+        log += item.title;
+        log += ", ";
+        log += item.thumbnail_path;
+        log += " },";
+      }
+      log += "] Actual: [";
+      for (const auto& item : sorted_actual) {
+        log += "{ ";
+        log += item.media_file_path;
+        log += ", ";
+        log += item.title;
+        log += ", ";
+        log += item.thumbnail_path;
+        log += " }";
+      }
+      log += "]";
+
+      FAIL() << log;
+    }
```

Comment:
I think this can be done better with custom gtest matchers.

https://github.com/google/googletest/blob/main/docs/advanced.md#teaching-googletest-how-to-print-your-values
https://github.com/google/googletest/blob/main/docs/reference/matchers.md#defining-matchers